src/cosmo.jl

```diff
@@ -87,35 +87,65 @@ end
+- `Kernel::AbstractArray{T, 2}`: A 2D array of type `T`, with dimensions `(n_bins, nχ)`. This stores kernel values for each tomographic bin and a grid of χ values.
 
+# Constructors
+- `GalaxyKernel{T}(n_bins::Int, nχ::Int)`: Creates a `GalaxyKernel` with the specified `n_bins` and `nχ` values, initializing the kernel values to zeros of type `T`.
+- `GalaxyKernel(n_bins::Int, nχ::Int)`: Creates a `GalaxyKernel` with the specified `n_bins` and `nχ` values, initializing the kernel values to zeros of type `Float64`.
 """
 @kwdef mutable struct GalaxyKernel{T} <: AbstractCosmologicalProbes{T}
-    Kernel::Vector{T} = zeros(500)
+    Kernel::AbstractArray{T, 2} = zeros(1, 1)
 end
 
+GalaxyKernel{T}(n_bins::Int, nχ::Int) where T = GalaxyKernel{T}(Kernel = zeros(T, n_bins, nχ))
+
+GalaxyKernel(n_bins::Int, nχ::Int) = GalaxyKernel{Float64}(n_bins, nχ)
+
 
 """
     ShearKernel{T} <: AbstractCosmologicalProbes{T}
 
-# Parameters:
-- `Kernel`: A vector of type `T` that holds the kernel values (default: zeros(500)).
+Represents a shear kernel used in cosmological lensing calculations. The kernel is defined over multiple tomographic bins.
+
+# Parameters
+- `Kernel::AbstractArray{T, 2}`: A 2D array of type `T`, with dimensions `(n_bins, nχ)`. Stores the kernel values for each tomographic bin and a grid of χ values.
+
+# Constructors
+- `ShearKernel{T}(n_bins::Int, nχ::Int)`: Creates a `ShearKernel` with the specified `n_bins` and `nχ` values, initializing the kernel values to zeros of type `T`.
+- `ShearKernel(n_bins::Int, nχ::Int)`: Creates a `ShearKernel` with the specified `n_bins` and `nχ` values, initializing the kernel values to zeros of type `Float64`.
 """
 @kwdef mutable struct ShearKernel{T} <: AbstractCosmologicalProbes{T}
-    Kernel::Vector{T} = zeros(500)
+    Kernel::AbstractArray{T, 2} = zeros(1, 1)
 end
 
+ShearKernel{T}(n_bins::Int, nχ::Int) where T = ShearKernel{T}(Kernel = zeros(T, n_bins, nχ))
+
+ShearKernel(n_bins::Int, nχ::Int) = ShearKernel{Float64}(n_bins, nχ)
+
 
 """
     CMBLensingKernel{T} <: AbstractCosmologicalProbes{T}
 
-# Parameters:
-- `Kernel`: A vector of type `T` that holds the kernel values (default: zeros(500)).
+Represents a CMB lensing kernel.
+
+# Parameters
+- `Kernel::AbstractArray{T, 2}`: A 2D array of type `T`, with dimensions `(n_bins, nχ)`. Note that CMB Lensing by definition only has a single tomographic bin.
+
+# Constructors
+- `CMBLensingKernel{T}(n_bins::Int, nχ::Int)`: Creates a `CMBLensingKernel` with the specified `n_bins` and `nχ` values, initializing the kernel values to zeros of type `T`.
+- `CMBLensingKernel(n_bins::Int, nχ::Int)`: Creates a `CMBLensingKernel` with the specified `n_bins` and `nχ` values, initializing the kernel values to zeros of type `Float64`.
 """
 @kwdef mutable struct CMBLensingKernel{T} <: AbstractCosmologicalProbes{T}
-    Kernel::Vector{T} = zeros(500)
+    Kernel::AbstractArray{T, 2} = zeros(1, 1)
```

Comment:
Is there a reason for this being a 2D array? Can't we force this to be a 1D array and then write a slightly different rule when computing, for instance, the $C_\ell$'s?

Reply:
I didn't change the $C_\ell$ computation but how the kernels are handled before.